app/views/play/level/modal/CourseVideosModalComponent.vue

```diff
@@ -0,0 +1,193 @@
+                span.rtl-allowed.video-title.gray-text
+                  | {{ level.title }}
+                p.video-desc.gray-text
+                  | {{ $t('courses.locked_videos_desc', { concept_name: level.title }) }}
+            .unlocked(v-else-if="level.videoStatus === 'unlocked'")
+              .video-link
+                img.video-image(
+                  :src='level.thumbnail_unlocked', 
+                  @click="onImageClick"
+                )
+                .video-status
+                  img.img-unlocked(src = '/images/pages/play/modal/unlocked_banner.png')
+              .video-text.m-t-2
+                span.rtl-allowed.video-title
+                  | {{ level.title }}
+                p.video-desc
+                  | {{ $t('courses.unlocked_videos_desc', { concept_name: level.title }) }}
+</template>
+
+<script>
+import api from 'core/api';
+import utils from 'core/utils'
+import Player from '@vimeo/player'
+
+export default Vue.extend({
+  props: {
+    courseInstanceID: {
+      type: String,
+      required: true
+    },
+    courseID: {
+      type: String,
+      required: true
+    }
+  },
+  data: () => ({
+    videoLevels: [],
+    vimeoPlayer: {}
+  }),
+  created() {
+    // fetch the levels that contain video URLs from the classroom ID for the given course instance ID.
+    api.courseInstances.get({courseInstanceID: this.courseInstanceID }).then(courseInstanceData => {
+      api.classrooms.get({classroomID: courseInstanceData.classroomID}).then(classroomData => {
+        const levels = classroomData.courses.find((c) => c._id == this.courseID).levels
+        for (let level in levels) {
+          const video = utils.videoLevels[levels[level].original]
+          if (video) {
+            this.videoLevels.push(video)
+          }
+        }
+        // fetch the sessions for the given user and course instance, and check if video levels are locked or unlocked.
+        api.courseInstances.getSessions({courseInstanceID: this.courseInstanceID }).then(sessions => {
+          let sessionsMap = {}
+          for (let s of sessions){
+            sessionsMap[s.level.original] = s
+          }
+          for (let l of this.videoLevels){
+            if (sessionsMap[l.original]){
+              Vue.set(l, 'videoStatus', 'unlocked')
+            }
+            else{
+              Vue.set(l, 'videoStatus', 'locked')
+            }
+          }
+        }).catch((err) => {
+          console.log("Error in fetching sessions:", err)
+        });
+      }).catch((err) => {
+        console.log("Error in fetching classroom details:", err)
+      });
+    }).catch((err) => {
+      console.log("Error in fetching course instance details:", err)
+    });
+  },
+  methods: {
+    // open video in the iframe when video thumbnail is clicked
+    onImageClick: function(e) {
+      let video_url = ''
+      let src = e.target.src
+      src = src.slice(src.search('/images'))
+      for (let l of this.videoLevels){
```

Comment:
`let video_url = (this.videoLevels.find(l => l.thumbnail_unlocked == src) || {}).url`?